coverage_comment/template.py

```diff
@@ -133,3 +140,21 @@ def pct(val: decimal.Decimal | float) -> str:
         return f"{val.quantize(decimal.Decimal('0.01'), rounding=decimal.ROUND_DOWN).normalize():f}%"
     else:
         return f"{val:.0%}"
+
+
+def get_file_url_function(repo_name: str, pr_number: int) -> Callable:
+    def _(
+        filename: pathlib.Path,
+        line_start: int | None = None,
+        line_end: int | None = None,
+    ) -> str:
+        # To link to a file in a PR, GitHub uses the link to the file overview combined with a SHA256 hash of the file path
+        s = f"https://github.com/{repo_name}/pull/{pr_number}/files#diff-{hashlib.sha256(str(filename).encode('utf-8')).hexdigest()}"
+
+        if line_start is not None:
+            s += f"R{line_start}"
+        if line_end is not None and line_end > line_start:
+            s += f"-R{line_end}"
```

Comment:
(might be worth adding as a comment that `R` stands for `Right` as in `Right side of the diff`. But given we compute coverage on the new code, we're only ever interested in the right side.)